src/Cli/dotnet/commands/dotnet-workload/list/VisualStudioWorkloads.cs

```diff
@@ -119,6 +122,36 @@ internal static class VisualStudioWorkloads
             }
         }
 
+        /// <summary>
+        /// Writes install records for VS Workloads so we later install the packs via the CLI for workloads managed by VS.
+        /// This is to fix a bug where updating the manifests in the CLI will cause VS to also be told to use these newer workloads via the workload resolver.
+        /// ...  but these workloads don't have their corresponding packs installed as VS doesn't update its workloads as the CLI does.
+        /// </summary>
+        internal static void WriteSDKInstallRecordsForVSWorkloads(IInstaller workloadInstaller, IWorkloadResolver workloadResolver,
+            IEnumerable<WorkloadId> workloadsWithExistingInstallRecords, IReporter reporter)
+        {
+            // Do this check to avoid adding an unused & unnecessary method to FileBasedInstallers
+            if (OperatingSystem.IsWindows() && typeof(NetSdkMsiInstallerClient) == workloadInstaller.GetType())
```

Comment:
Is there a reason we couldn't call `WorkloadInstallType.GetWorkloadInstallType()` here instead of doing a .NET type comparison?